nixos/modules/config/networking.nix

```diff
@@ -213,6 +221,12 @@ in
               otherHosts = allToString ( removeAttrs cfg.hosts [ "127.0.0.1" "::1" ]);
           in
           ''
+            ${optionalString cfg.hostnameInHosts ''
+              127.0.1.1 ${cfg.hostName}
```

Comment:
Note that it can be empty: https://github.com/NixOS/nixpkgs/blob/release-17.09/nixos/modules/tasks/network-interfaces.nix#L326-L327